packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts

```diff
@@ -1,7 +1,28 @@
 import { defineAuth } from '@aws-amplify/backend';
 
+/**
+ * Define and configure your auth resource
+ * When used alongside data, it is automatically configured as an auth provider for data
+ * @see https://docs.amplify.aws/gen2/build-a-backend/auth
+ */
 export const auth = defineAuth({
   loginWith: {
     email: true,
+    // add social providers
+    externalProviders: {
+      // first, create your secrets using `amplify sandbox secret`
+      // then, import `secret` from `@aws-amplify/backend`
+      // loginWithAmazon: {
+      //   clientId: secret('LOGINWITHAMAZON_CLIENT_ID'),
+      //   clientSecret: secret('LOGINWITHAMAZON_CLIENT_SECRET'),
+      // }
+    },
```

Comment:
I'd also add MFA as when enabled with SMS, it requires the same prerequisites for sending SMS messages via Amazon SNS.

Reply:
added